File: 0231-power-of-two/0231-power-of-two.java
class Solution {
     public boolean isPower(int n){
         if(n==0){
             return false;
         }
         if(n==1){return true;}
         if((n%2==0 && isPowerOfTwo(n/2))){
             return true;
         }
         return false;
     }
    public boolean isPowerOfTwo(int n) {
        return isPower(n);
    }
}

File: 0231-power-of-two/0231-power-of-two.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isPower = function(n){
    if(n== 1){
            return true;
        }
        if(n==0){
            return false;
        }
        if(n%2==0){
            if(isPower(n/2)){
                return true;
            }
        }
        return false;
} 
var isPowerOfTwo = function(n) {
     return isPower(n);
};